code/ship/ship.cpp

```diff
@@ -6718,7 +6718,8 @@ static void ship_find_warping_ship_helper(object *objp, dock_function_info *info
 		return;
 
 	// am I arriving or departing by warp?
-	if (Ships[objp->instance].is_arriving() || Ships[objp->instance].flags[Ship_Flags::Depart_warp])
+	// ignore _ndl flags as they are now on all docked-warping ships; we need the dock leader!
+	if (Ships[objp->instance].flags[Ship_Flags::Arriving_stage_1, Ship_Flags::Arriving_stage_2, Ship_Flags::Depart_warp])
```

Comment:
Couldn't the same thing be done by adding an argument to `is_arriving()`, rather than by hardcoding this list of flags every time it's needed?

Reply:
fair call; I'll see about updating that on the weekend. How do you feel about a separate inline function rather than an tri-state argument to `is_arriving()`? e.g. `is_arriving_dock_leader()` & `is_arriving_stage_1_all_docked()`

Reply:
As long as we're not using a hardcoded flag list in multiple places, I don't particularly mind implementation.